apps/chat/components/assistant-ui/thread.tsx

```diff
@@ -0,0 +1,260 @@
+import {
+  ActionBarPrimitive,
+  ComposerPrimitive,
+  MessagePrimitive,
+  ThreadPrimitive,
+  useMessage,
+} from '@assistant-ui/react'
+import {
+  ArrowDownIcon,
+  CheckIcon,
+  CopyIcon,
+  PencilIcon,
+  RefreshCwIcon,
+  SendHorizontalIcon,
+} from 'lucide-react'
+import type { FC } from 'react'
+
+import { BranchPicker } from '@/components/assistant-ui/branch-picker'
+import { MarkdownText } from '@/components/assistant-ui/markdown-text'
+import { TooltipIconButton } from '@/components/assistant-ui/tooltip-icon-button'
+import { Button } from '@/components/ui/button'
+import { ToolFallback } from './tool-fallback'
+
+export const Thread: FC = () => {
+  return (
+    <ThreadPrimitive.Root
+      className="bg-background box-border flex h-full flex-col overflow-hidden"
+      style={{
+        ['--thread-max-width' as string]: '42rem',
+      }}
+    >
+      <ThreadPrimitive.Viewport className="flex h-full flex-col items-center overflow-y-scroll scroll-smooth bg-inherit px-4 pt-8">
+        <ThreadWelcome />
+
+        <ThreadPrimitive.Messages
+          components={{
+            UserMessage: UserMessage,
+            EditComposer: EditComposer,
+            AssistantMessage: AssistantMessage,
+          }}
+        />
+
+        <ThreadPrimitive.If empty={false}>
+          <div className="min-h-8 flex-grow" />
+        </ThreadPrimitive.If>
+
+        <div className="sticky bottom-0 mt-3 flex w-full max-w-[var(--thread-max-width)] flex-col items-center justify-end rounded-t-lg bg-inherit pb-4">
+          <ThreadScrollToBottom />
+          <Composer />
+        </div>
+      </ThreadPrimitive.Viewport>
+    </ThreadPrimitive.Root>
+  )
+}
+
+const ThreadScrollToBottom: FC = () => {
+  return (
+    <ThreadPrimitive.ScrollToBottom asChild>
+      <TooltipIconButton
+        tooltip="Scroll to bottom"
+        variant="outline"
+        className="absolute -top-8 rounded-full disabled:invisible"
+      >
+        <ArrowDownIcon />
+      </TooltipIconButton>
+    </ThreadPrimitive.ScrollToBottom>
+  )
+}
+
+const ThreadWelcome: FC = () => {
+  return (
+    <ThreadPrimitive.Empty>
+      <div className="flex w-full max-w-[var(--thread-max-width)] flex-grow flex-col">
+        <div className="flex w-full flex-grow flex-col items-center justify-center">
+          <p className="mt-4 font-medium">How can I help you today?</p>
+        </div>
+        <ThreadWelcomeSuggestions />
+      </div>
+    </ThreadPrimitive.Empty>
+  )
+}
+
+const ThreadWelcomeSuggestions: FC = () => {
```

Comment:
are these actually shown to the user? if yes, we should think about how we can make these more generic to match the course chatbot use case